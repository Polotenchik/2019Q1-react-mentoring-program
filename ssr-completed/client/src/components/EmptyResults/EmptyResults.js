import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Theme from '../../common/Theme/Theme';

const EmptyResults = ({ text }) => (
  <EmptyBackground>
    <EmptyContainer>
      <EmptyText>{text}</EmptyText>
    </EmptyContainer>
  </EmptyBackground>
);

export default EmptyResults;

EmptyResults.propTypes = {
  text: PropTypes.string.isRequired,
};

const EmptyBackground = styled.div`
    flex: auto;
    width: 100%;
    height: auto;
    background-color: ${Theme.colors.white};
`;

const EmptyContainer = styled.div`
    width: 80%;
    margin: 0 auto;  
`;

const EmptyText = styled.p.attrs({
  className: 'empty-text',
})`
   color: ${Theme.colors.red}; 
   text-align: center;
   vertical-align: middle;
   font-size: 30px;
`;
