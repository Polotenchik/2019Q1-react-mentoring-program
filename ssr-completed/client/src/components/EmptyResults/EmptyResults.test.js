import React from 'react';
import { shallow } from 'enzyme';
import EmptyResults from './EmptyResults';

describe('EmptyResults Snapshot', () => {
    it('EmptyResults Component should render correctly with given props', () => {
        const text='Empty result';
        const component = shallow(<EmptyResults text={text} />);
        expect(component).toMatchSnapshot();
    });
});

