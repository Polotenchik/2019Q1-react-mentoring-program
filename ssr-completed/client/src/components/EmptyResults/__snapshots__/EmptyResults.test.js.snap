// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyResults Snapshot EmptyResults Component should render correctly with given props 1`] = `
<styled.div>
  <styled.div>
    <styled.p>
      Empty result
    </styled.p>
  </styled.div>
</styled.div>
`;
